libqtile/backend/wayland/core.py

```diff
@@ -108,6 +108,39 @@
     from libqtile.core.manager import Qtile
 
 
+class ImplicitGrab:
+    """Keep track of an implicit pointer grab.
+
+    A Wayland client expects to receive pointer events from the moment a
+    pointer button is pressed on its surface until the moment the button is
+    released.  The Wayland protocol leaves this behavior to the compositor.
+    """
+
+    def __init__(
+        self,
+        core: Core,
+        surface: Surface,
+        start_x: int,
+        start_y: int,
+        start_sx: int,
+        start_sy: int,
+    ) -> None:
+        self.core = core
+        self.surface = surface
+        self.start_x = start_x
+        self.start_y = start_y
+        self.start_sx = start_sx
+        self.start_sy = start_sy
```

Comment:
It looks like the two x values and two y values are only used together, so maybe we could collapse them. E.g.:
```suggestion
        self.start_dx = start_sx - start_x
        self.start_dy = start_sy - start_y
```
Then just use that one below in `_implicit_grab_motion`

Reply:
Good point!  I was just copying the computation from Sway.